pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go

```diff
@@ -1364,3 +1372,51 @@ func isOwnedByFluentOperator(obj client.Object) bool {
 	}
 	return false
 }
+
+// WaitUntilRequiredExtensionsReady checks and waits until all required extensions for a seed exist and are ready.
+func WaitUntilRequiredExtensionsReady(ctx context.Context, gardenClient client.Client, seed *gardencorev1beta1.Seed, interval, timeout time.Duration) error {
+	return retry.UntilTimeout(ctx, interval, timeout, func(ctx context.Context) (done bool, err error) {
+		if err := RequiredExtensionsReady(ctx, gardenClient, seed); err != nil {
+			return retry.MinorError(err)
+		}
+
+		return retry.Ok()
+	})
+}
+
+// RequiredExtensionsReady checks if all required extensions for a seed exist and are ready.
+func RequiredExtensionsReady(ctx context.Context, gardenClient client.Client, seed *gardencorev1beta1.Seed) error {
+	controllerInstallationList := &gardencorev1beta1.ControllerInstallationList{}
+	if err := gardenClient.List(ctx, controllerInstallationList, client.MatchingFields{
+		core.SeedRefName: seed.Name,
+	}); err != nil {
+		return err
+	}
+
+	requiredExtensions := gardenerutils.ComputeRequiredExtensionsForSeed(seed)
+
+	for _, controllerInstallation := range controllerInstallationList.Items {
+		controllerRegistration := &gardencorev1beta1.ControllerRegistration{}
+		if err := gardenClient.Get(ctx, client.ObjectKey{Name: controllerInstallation.Spec.RegistrationRef.Name}, controllerRegistration); err != nil {
+			return err
+		}
+
+		for _, kindType := range requiredExtensions.UnsortedList() {
+			split := strings.Split(kindType, "/")
+			if len(split) != 2 {
+				return fmt.Errorf("unexpected required extension: %q", kindType)
+			}
+			extensionKind, extensionType := split[0], split[1]
+
+			if helper.IsResourceSupported(controllerRegistration.Spec.Resources, extensionKind, extensionType) && helper.IsControllerInstallationSuccessful(controllerInstallation) {
+				requiredExtensions.Delete(kindType)
+			}
+		}
+	}
+
+	if len(requiredExtensions) > 0 {
+		return fmt.Errorf("extension controllers missing or unready: %+v", requiredExtensions)
+	}
+
+	return nil
+}
```

Comment:
This looks too similar to https://github.com/gardener/gardener/blob/13d5158c55cc6b21ff1a24bb068c00dd223e8563/pkg/operation/botanist/botanist.go#L240-L282. Can you find a way to make the code reusable?